src/utils/common.ts

```diff
@@ -20,7 +20,15 @@ export function getErrorMessage(value: unknown): string {
     }
 
     if (isRouteErrorResponse(value)) {
-        return value.data;
+        if (typeof value.data == 'object' && value.data !== null) {
```

Comment:
How did it allow you to return `value.data` before? Is it `any`?

Reply:
it's `any`. I suspect that during server-side rendering ErrorResponse['data'] in error boundary is of that type that was thrown in loader. But on the client it's serialised to string.

Reply:
```
export const loader = () => {
    throw new json({code: 'ERROR_CODE', message: 'Something went wrong'});
}

export const Page = () => {
  const data = useLoaderData()
}

export const ErrorBoundary = () => {
    const error = useRouteError();
    if (isRouteErrorResponse(error)) {
        typeof error.data // "object" on the server, "string" on the client
    }
}

```

Reply:
no... disregard what I wrote above

Reply:
1. `throw json({ code: 'ERROR_CODE', message: 'Something went wrong' })` in loader, `error.data` is going to be an object in error boundary (on server and on client)

2. `throw new Response('Something went wrong')` in loader, `error.data` is going to be a string